### ISSUE TITLE
Decouple refill from capacity

### ISSUE THREAD

**Issue by efenderbosch** [vladimir-bukhtoyarov/bucket4j#13]
https://help.shopify.com/api/getting-started/api-call-limit

> The bucket size is 40 calls (which cannot be exceeded at any given time), with a "leak rate" of 2 calls per second that continually empties the bucket. If your app averages 2 calls per second, it will never trip a 429 error ("bucket overflow").

Other APIs have similar burstable capacity. Other Java "leaky bucket" implementations support burstable capacity, but the fact that bucket4j supports distributed tokens through JCache is very interesting.

**Comment by vladimir-bukhtoyarov** [comment]
[The token-bucket library is good example](https://github.com/bbeck/token-bucket#usage) how to this issue can be done.

With token-library you can achieve the requirements above:
```java
TokenBucket bucket = TokenBuckets.builder()
       .withCapacity(40)
      .withFixedIntervalRefillStrategy(1, 500, TimeUnit.MILLISECONDS)
      .build();
```
These requirements can not be solved by Bucket4j because of ```refill``` is not independent entity, speed of refill is tightly coupled with bandwidth capacity. 

But it is very trivial job to decouple refill from capacity. I suppose to use following API for case when refill and capacity should be independent:
```java
Bucket bucket = Bucket4j.builder()
      withLimitedBandwidth(40, Refill.fixed(2, Duration.ofSeconds(1)))
      .build();
```
The old builder API should stay unchanged, because from my experience the situations when refill and capacity should be coupled are more frequent.

@efenderbosch do you have any concerns according this design?

**Comment by efenderbosch** [comment]
Looks good, I like that builder API.

**Comment by vladimir-bukhtoyarov** [comment]
The coding and testing is done. Functionality will be released in few days in scope of release 1.3(after #16 will be completed).

Example of usage can be found [there](https://github.com/vladimir-bukhtoyarov/bucket4j/blob/1.3/doc-pages/basic-usage.md#example-3---limiting-the-rate-of-access-to-rest-api).
